classes/metric/base.php

```diff
@@ -131,4 +131,23 @@ public function get_settings_url() {
      * @return metric_item
      */
     abstract public function get_metric_item(): metric_item;
+
+    /**
+     * Metric's ability to be backfilled.
+     *
+     * @return bool
+     */
+    public function is_backfillable(): bool {
+        return false;
+    }
+
+    /**
+     * Returns records for backfilled metric.
+     *
+     * @param array $data Array of data specifying timeframe and sample to backfill.
+     * @return array|null
+     */
+    public function get_saved_metric_data(array $data): ?array {
```

Comment:
Can I suggest a name tweak like get_metric_items or get_metric_items_for_period?

I also dislike the $data argument, this would be much more readable as explicit arguments

Reply:
Fixed